source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -18,109 +20,197 @@ minimize the objective function
+      int64_t get_max_iteration_count() const;
       double get_accuracy_threshold() const;
-      std::int64_t get_cluster_count() const;
-      std::int64_t get_max_iteration_count() const;
 
-      desc& set_gamma(double);
+      desc& set_cluster_count(int64_t);
+      desc& set_max_iteration_count(int64_t);
       desc& set_accuracy_threshold(double);
-      desc& set_cluster_count(std::int64_t);
-      desc& set_max_iteration_count(std::int64_t);
    };
 
-   } // namespace onedal::kmeans
+cluster_count ``int64_t`` ``default = 2``
+   The number of clusters.
+
+   Invariants
+      | ``cluster_count > 0``
+
+max_iterations ``int64_t`` ``default = 100``
+   The maximum number of iterations.
+
+   Invariants
+      | ``max_iterations > 0``
+
+accuracy_threshold ``double`` ``default = 0.0``
+   The threshold for termination.
+
+   Invariants
+      | ``accuracy_threshold >= 0.0``
 
 
 Model
 -----
-.. code-block:: c++
+::
 
    class model {
    public:
       const table& get_centroids() const;
+      int64_t get_cluster_count() const;
    };
 
+centroids ``table`` ``[read-only]``
+   :math:`[\mathrm{cluster\_count} \times \mathrm{feature\_count}]` table with the
+   cluster centroids. Each row of the table stores one centroid.
+
+   Invariants
+      | ``centroids.is_empty == false``
+
+cluster_count ``int64_t`` ``[read-only]``
+   Number of clusters in the trained model.
 
-Training
---------
+   Invariants
+      | ``cluster_count == centroids.row_count``
 
+
+Training ``onedal::train(...)``
+-------------------------------
 Input
 ~~~~~
-.. code-block:: c++
-
-   namespace onedal::kmeans {
+::
 
    class train_input {
    public:
       train_input();
       train_input(const table& data);
       train_input(const table& data, const table& initial_centroids);
 
+      const table& get_data() const;
+      const table& get_initial_centroids() const;
+
       train_input& set_data(const table&);
       train_input& set_initial_centroids(const table&);
    };
 
-   } // namespace onedal::kmeans
+data ``table`` ``default = table()``
+   :math:`[\mathrm{observation\_count} \times \mathrm{feature\_count}]` table with the data
+   to be clustered, where each row stores one observation.
 
+initial_centroids ``table`` ``default = table()``
+   :math:`[\mathrm{cluster\_count} \times \mathrm{feature\_count}]` table with the
+   initial centroids, where each row stores one centroid.
 
 Result
 ~~~~~~
-.. code-block:: c++
-
-   namespace onedal::kmeans {
+::
 
    class train_result {
    public:
       const model& get_model() const;
       const table& get_labels() const;
-      std::int64_t get_iteration_count() const;
+      int64_t get_iteration_count() const;
       double get_objective_function_value() const;
    };
 
-   } // namespace onedal::kmeans
+model ``kmeans::model`` ``[read-only]``
+   The trained K-means model.
 
+labels ``table`` ``[read-only]``
+   :math:`[\mathrm{observation\_count} \times 1]` table with assignments of cluster
```

Comment:
table size is required
labels vs cluster indices - minor inconsistency. Need to introduce and fix the terminology across the section